browser/brave_wallet/brave_wallet_provider_delegate_impl.cc

```diff
@@ -171,8 +173,61 @@ void BraveWalletProviderDelegateImpl::IsAccountAllowed(
       base::BindOnce(&OnIsAccountAllowed, account, std::move(callback)));
 }
 
+void BraveWalletProviderDelegateImpl::AddSolanaConnectedAccount(
+    const std::string& account) {
+  if (!web_contents_)
+    return;
+  auto* tab_helper =
+      brave_wallet::BraveWalletTabHelper::FromWebContents(web_contents_);
+  if (tab_helper)
+    tab_helper->AddSolanaConnectedAccount(host_id_, account);
+}
+
+void BraveWalletProviderDelegateImpl::RemoveSolanaConnectedAccount(
+    const std::string& account) {
+  if (!web_contents_)
+    return;
+  auto* tab_helper =
+      brave_wallet::BraveWalletTabHelper::FromWebContents(web_contents_);
+  if (tab_helper)
+    tab_helper->RemoveSolanaConnectedAccount(host_id_, account);
+}
+
+bool BraveWalletProviderDelegateImpl::IsSolanaAccountConnected(
+    const std::string& account) {
+  if (!web_contents_)
+    return false;
+  auto* tab_helper =
+      brave_wallet::BraveWalletTabHelper::FromWebContents(web_contents_);
+  if (!tab_helper)
+    return false;
+
+  return tab_helper->IsSolanaAccountConnected(host_id_, account);
+}
+
 void BraveWalletProviderDelegateImpl::WebContentsDestroyed() {
   web_contents_ = nullptr;
 }
 
+void BraveWalletProviderDelegateImpl::RenderFrameHostChanged(
+    content::RenderFrameHost* old_host,
+    content::RenderFrameHost* new_host) {
+  if (!old_host)
+    return;
+  auto* tab_helper =
+      brave_wallet::BraveWalletTabHelper::FromWebContents(web_contents_);
+  if (tab_helper)
+    tab_helper->ClearSolanaConnectedAccounts(old_host->GetGlobalId());
```

Comment:
Should probably use host_id_ here and below to make sure we always only clear on the correct render frame, which is the one that is passed in when constructing this.

Reply:
resolved in 526197b3ddd94a11e9abdcb9494a365e5bfdb912